logstash-core/lib/logstash/pipeline.rb

```diff
@@ -34,6 +34,7 @@ module LogStash; class BasePipeline
 
   def initialize(config_str, settings = SETTINGS)
     @logger = self.logger
+    @mutex = Mutex.new
```

Comment:
suggestion: we could use a more descriptive identifier? maybe something like `@write_client_mutex` ?